File: bin/Data/EditorLua/view/Inspector.lua
local Inspector = {}
Inspector.isShow = true
local Translator = require("EditorLua/core/Translate")
local TransformIns = require("EditorLua/view/inspectors/TransformInspector")
local AniModelIns = require("EditorLua/view/inspectors/AniModelInspector")
Inspector.nodeEnable = true

function Inspector:Update()
    TransformIns.app = self.app
    if self.isShow == true then
        self.isShow = imgui.Begin(Translator:translate("Inspector"),self.isShow)
        if self.app.cppApp:GetSelectNode() ~= nil then
            self.nodeEnable = self.app.cppApp:GetSelectNode():IsEnabled()
            local tmpBool
            tmpBool,self.nodeEnable = imgui.Checkbox("Enable",self.nodeEnable)
            TransformIns:Update()
            local aniModel = self.app.cppApp:GetSelectNode():GetComponent("AnimatedModel",false)
            if aniModel ~= nil then
                AniModelIns:Update()
            end
            if self.nodeEnable ~= self.app.cppApp:GetSelectNode():IsEnabled() then
                --self.app.cppApp:MakeCurent()
                self.app.cppApp:GetSelectNode():SetEnabled(self.nodeEnable)
            end
        end
        imgui.End()
    end
end

function Inspector:Show(show)
    self.isShow = show
end

return Inspector

File: Source/UrhoEditor/views/Inspector.cpp
#include "Inspector.h"
#include "EditorApp.h"
#include <Urho3D/Graphics/AnimatedModel.h>
#include "views/inspectors/VariantDrawer.h"

namespace Urho3DEditor 
{
Inspector::Inspector() 
{
	_transformIns = new TransformInspector();
	_aniModelIns = new AniModelInspector();
}
Inspector::~Inspector() 
{
}
void Inspector::Update() 
{
	if(!showing)
		return;
	Node* selectedNode = EditorApp::GetInstance()->GetSelectNode();
	ImGui::Begin("Inspector",&showing);
	if(selectedNode)
	{
		_nodeEnable = selectedNode->IsEnabled();
		VariantDrawer::DrawBool("  Enable", _nodeEnable);
		_transformIns->Update();
		if(selectedNode->HasComponent<AnimatedModel>())
		{
			_aniModelIns->Update();
		}
		if(_nodeEnable != selectedNode->IsEnabled())
		{
			selectedNode->SetEnabled(_nodeEnable);
		}
	}
	ImGui::End();

}
}